Automattic-Tracks-iOS/Crash Logging/CrashLogging.swift

```diff
@@ -2,178 +2,251 @@ import Foundation
 import Sentry
 import CocoaLumberjack
 
+struct CrashLoggingInternals {
+    static var crashLogging: CrashLogging?
+}
+
 /// A class that provides support for logging crashes. Not compatible with Objective-C.
 public class CrashLogging {
 
-    /// A singleton is maintained, but the host application needn't be aware of its existence.
-    internal static let sharedInstance = CrashLogging()
-
-    private var dataProvider: CrashLoggingDataProvider?
-    private var eventLogging: EventLogging?
+    private let dataProvider: CrashLoggingDataProvider
+    private let eventLogging: EventLogging?
 
-    /// Thread-safe single initialization
-    fileprivate static let threadSafeDispatchQueue = DispatchQueue(label: Bundle.main.bundleIdentifier ?? "tracks" + "-crash-logging-queue")
-    fileprivate static var _isStarted = false
-    internal static var isStarted: Bool {
-        get {
-            return threadSafeDispatchQueue.sync { _isStarted }
-        }
-        set {
-            threadSafeDispatchQueue.sync { _isStarted = newValue }
-        }
+    public init(dataProvider: CrashLoggingDataProvider, eventLogging: EventLogging? = nil) {
+        self.dataProvider = dataProvider
+        self.eventLogging = eventLogging
     }
 
-    /**
-     Initializes the crash logging system.
-
-     - Parameters:
-     - dataProvider: An object that will provide any required data to the crash logging system.
-
-     - SeeAlso: CrashLoggingDataProvider
-     */
-    public static func start(withDataProvider dataProvider: CrashLoggingDataProvider, eventLogging: EventLogging? = nil) {
+    public func start() throws -> CrashLogging {
```

Comment:
`throws` because the DSN may be invalid, and we don't want to hide that error – otherwise we might ship an app to production with no ability to track crashes